boxsdk/util/api_call_decorator.py

```diff
@@ -0,0 +1,53 @@
+# coding: utf-8
+
+from __future__ import unicode_literals, absolute_import
+
+from functools import update_wrapper
+
+
+def api_call(func):
+    """
+    Designates the decorated method as one that makes a Box API call.
+    The decorated method can then accept a new keyword argument `request_data`,
+    a dictionary of key-value pairs to be passed to the network layer for API
+    calls made by the method.
+
+    :param func:
+        The method to decorate.
+    :type func:
+        `callable`
+    :return:
+        A wrapped method that can pass extra request data to the network layer.
+    :rtype:
+        :class:`APICallWrapper`
+    """
+    wrapper = APICallWrapper(func)
+    update_wrapper(wrapper, func)
+    return wrapper
+
+
+class APICallWrapper(object):
+
+    def __init__(self, func_that_makes_an_api_call):
+        super(APICallWrapper, self).__init__()
+        self._func_that_makes_an_api_call = func_that_makes_an_api_call
+
+    def __get__(self, _instance, owner):
+        def call(*args, **kwargs):
```

Comment:
This will cause us to need to construct a new function every time we want to make an API call.

To be more economical, we could try something like this:

```python
class APICallWrapper(object):
    def __init__(self, func_that_makes_an_api_call, instance=None, owner=None):
        ...
        self._instance = instance
        self._owner = owner

    def __get__(self, instance, owner):
        return self.__class__(func_that_makes_an_api_call, instance=instance, owner=owner)

    def __call__(*args, **kwargs):
        ...
```

As another idea, we can consider using https://pypi.python.org/pypi/wrapt to power the decorator, so that we don't have to think at all about the descriptor. Though that would introduce its own slowness.

If we keep this code as-is for now, we should use `@wraps` on the definition of `call`.